src/classes/CRLP_RollupUI_SVC.cls

```diff
@@ -0,0 +1,347 @@
+        model.filterGroups = getFilterGroupDefinitions();
+
+        //serialize to avoid passing inner classes, which is technically unsupported
+        String modelString = JSON.serialize(model);
+        return modelString;
+    }
+
+    /*******************************************************************************************************
+    * @description Queries from the rollup selector to fetch all rollup__mdt records. Parses so that only the relevant
+    * information is returned, and a custom RollupTableItem class is created with this information.
+    * @return a list of all rollup definitions
+    */
+    @AuraEnabled
+    public static List<RollupTableItem> getRollupDefinitions() {
+        List<RollupTableItem> mdtTableList = new List<RollupTableItem>();
+
+        List<Rollup__mdt> mdtList = CRLP_Rollup_SEL.getRollupDefinitions();
+
+        for(Integer i=0; i<mdtList.size(); i++){
+            RollupTableItem item = new RollupTableItem();
+
+            if(mdtList[i].Summary_Object__r.QualifiedAPIName == 'Account' && mdtList[i].Detail_Object__r.QualifiedAPIName == 'Opportunity') {
+                item.creditType = Label.CRLP_HardCredit;
+            } else if(mdtList[i].Summary_Object__r.QualifiedAPIName == 'Account' && mdtList[i].Detail_Object__r.QualifiedAPIName == 'npe01__OppPayment__c') {
+                item.creditType = Label.CRLP_HardCredit;
+            } else if(mdtList[i].Summary_Object__r.QualifiedAPIName == 'Account' && mdtList[i].Detail_Object__r.QualifiedAPIName == 'Partial_Soft_Credit__c') {
+                item.creditType = Label.CRLP_SoftCredit;
+            } else if(mdtList[i].Summary_Object__r.QualifiedAPIName == 'Contact' && mdtList[i].Detail_Object__r.QualifiedAPIName == 'Opportunity') {
+                item.creditType = Label.CRLP_HardCredit;
+            } else if(mdtList[i].Summary_Object__r.QualifiedAPIName == 'Contact' && mdtList[i].Detail_Object__r.QualifiedAPIName == 'Partial_Soft_Credit__c') {
+                item.creditType = Label.CRLP_SoftCredit;
+            } else if(mdtList[i].Summary_Object__r.QualifiedAPIName == 'Contact' && mdtList[i].Detail_Object__r.QualifiedAPIName == 'npe01__OppPayment__c') {
+                item.creditType = Label.CRLP_HardCredit;
+            } else if(mdtList[i].Summary_Object__r.QualifiedAPIName == 'General_Accounting_Unit__c' && mdtList[i].Detail_Object__r.QualifiedAPIName == 'Allocation__c') {
+                item.creditType = Label.CRLP_HardCredit;
+            } else {
+                item.creditType = Label.CRLP_HardCredit;
+            }
+            item.detailObject = mdtList[i].Detail_Object__r.Label;
+            item.summaryObject = mdtList[i].Summary_Object__r.Label;
+            item.rollupName = mdtList[i].MasterLabel;
+            item.detailField = mdtList[i].Detail_Field__r.Label;
+            item.filterGroupName = mdtList[i].Filter_Group__r.MasterLabel;
+            item.operation = mdtList[i].Operation__c.replace('_', ' ');
```

Comment:
Reviewing this method, rollupOperationTypesByName is using hard-coded strings which would not be translatable if we used this method. Should we change CRLP_Operation to replace the strings like we have here, or is there another way you'd prefer to get around this issue?

Reply:
Very good point. Same with getAvailableYearlyOperationTypes(). Looks like it's possible to use `Rollup__mdt.Operation__c.getDescribe().getPicklistValues()`, so it would make sense to refactor that class to loop through this collection to create the Map. The picklist api name will always be the same (ex: "YEARS_DONATED"), which is needed to match to the Enum's, but the label changes based on translations. 

This will apply to CMT_FilterRuleUI_SVC.getFilterOperations() as well. Out of curiosity, are you using CMT_FilterRuleUI_SVC.getAvailableOperations() to retrieve the appropriate filter rule operation types based on object and field, or is that now in JS only?

Reply:
We're getting available operations in the CRLP_RollupUI_SVC.getYearlyOperations() and .getFilterGroupDefinitions(), and getting the picklist values + labels is exactly what we're using, but if that would be better to put in the CMT_FilterRuleUI_SVC that's ok too.